### PR TITLE
Use address as title when there is an address and no title

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Ugh. Boom. Not being able to run the tests locally is kinda annoying. No time to look into this now tho. Probably its just integration tests that need adjustment

**Comment by PeterTheOne** [comment]
> Probably its just integration tests that need adjustment.

Yes.

File: tests/phpunit/parsers/LocationParserTest.php [hunk 1]
```diff
@@ -134,4 +134,22 @@ protected function getInstance() {
 		return new LocationParser();
 	}
 
+	public function testGivenAddressAndNoTitle_addressIsSetAsTitle() {
+		$location = ( new LocationParser() )->parse( 'Tempelhofer Ufer 42' );
+
+		$this->assertSame( 'Tempelhofer Ufer 42', $location->getTitle );
```
Comment (JeroenDeDauw) [inline tests/phpunit/parsers/LocationParserTest.php]: What the heck. I forgot the function call brackets? :D

**Comment by JeroenDeDauw** [comment]
I cannot easily finish this as I'm not able to run the tests locally. So if you'd like to see this happen, feel free to pick this up.

**Comment by PeterTheOne** [comment]
I did some work on the unit tests but

    <display_map searchmarkers="all">
    New York City
    Brussels
    Berlin
    London
    Norway
    </display_map>

doesn't should the address as titles on the map for me.

**Comment by JeroenDeDauw** [comment]
It's not nice to have tests bind to LocationParser, esp if the production code they test does not directly bind to it.

How about adding an alternate constructor to Location? Something like `Location::newTitledLocation( $latitude, $longitude )`

Not that many tests should be affected by this change in LocationParser anyway. If they are, that suggests a problem somewhere. Perhaps some should just assert lat and long, and not care about the rest of the Location. I'm not sure as I have not really looked at the involved code.

**Comment by PeterTheOne** [comment]
> How about adding an alternate constructor to Location? Something like Location::newTitledLocation( $latitude, $longitude )

I did this in 38b79f1.

> Not that many tests should be affected by this change in LocationParser anyway. If they are, that suggests a problem somewhere. Perhaps some should just assert lat and long, and not care about the rest of the Location. I'm not sure as I have not really looked at the involved code.

Would be the cleanest solution, but I don't want to change https://github.com/JeroenDeDauw/Maps/blob/master/tests/phpunit/parserhooks/ParserHookTest.php#L88 anytime soon.

**Comment by JeroenDeDauw** [comment]
Right, for that code I'd just use this new alternative constructor

**Comment by JeroenDeDauw** [comment]
Pffft, this Location class is foobar. Even though it was way worse before the last refactor I did to it. No reason for this thing to be mutable. It should just take all its stuff in its constructor. And not have an alternative one that does geocoding based on global state services >_>

**Comment by JeroenDeDauw** [comment]
Random rant about something else aside, this PR looks good now.

I'm a bit worried that this behavior change might not be desired for some people though. What if you want to have a map with a bunch of markers but no pop-ups? When using addresses, all of a sudden after updating Maps, you'll get pop-ups. And you won't be able to turn it off. (Or do we have a general off setting for those things?) Thoughts @kghbln?

Therefore I do not want to merge this just yet, as then It'd be released in a few days. I'd like to ask the users for feedback first. This could be done as part of the upcoming release mail.

**Comment by kghbln** [comment]
> What if you want to have a map with a bunch of markers but no pop-ups? When using addresses, all of a sudden after updating Maps, you'll get pop-ups. And you won't be able to turn it off.

In this case it would be preferable to have a query parameter allowing to say yes or no.

**Comment by PeterTheOne** [comment]
Should we name it (frame it) like `hidepopups` or `useaddressastitle`?

**Comment by JeroenDeDauw** [comment]
Good question. Btw, I suspect the maps docs are not using the word popup, but rather something like "info windows". Should probably be consistent with the stuff users actually see.

**Comment by PeterTheOne** [comment]
Ok, I'm in favor of something similar to `hidepopups`..

**Comment by kghbln** [comment]
What about just `popups` with "yes/no" or (`addresstitle`) with "yes/no". Hmm..., prefer the latter since `popups` seems to be too generic. `hidepopups` or `useaddressastitle` aleady contain sort of a "yes" or "no"  indicating of the default setting which only a few other options to "Maps" do. Since we also have a `title` option this appears to be more consistent to me.

**Comment by PeterTheOne** [comment]
One month later.. so how should we proceed. :P

**Comment by kghbln** [comment]
@JeroenDeDauw Ping :P

**Comment by JeroenDeDauw** [comment]
Having a `useaddressastitle` parameter is fine by me

File: includes/parsers/LocationParser.php [hunk 2]
```diff
@@ -33,6 +43,8 @@ class LocationParser extends StringValueParser {
 	public function stringParse( $value ) {
 		$separator = '~';
 
+		$useaddressastitle = $this->getOption('useaddressastitle');
```
Comment (JeroenDeDauw) [inline includes/parsers/LocationParser.php]: `$useAddressAsTitle = $this->getOption( 'useaddressastitle' );`

**Comment by JeroenDeDauw** [comment]
Thanks!

**Comment by kghbln** [comment]
Just comes to my mind: Shouldn't we also get an entry for `useaddressastitle` to show via the `#mapsdoc` parserfunction? I guess this option applies to all services.

**Comment by PeterTheOne** [comment]
For now this only added an `useaddressastitle` option for the LocationParser, correct. I'm not sure where I would set this in the maps parser function.

**Comment by kghbln** [comment]
I thought this is going to be an option to the "display_map" parser function. In this case it should probably go to the [services files](https://github.com/JeroenDeDauw/Maps/tree/master/includes/services) like "Maps_Leaflet.php" etc. like the other options. I may be wrong though. @JeroenDeDauw ?

**Comment by JeroenDeDauw** [comment]
If it is not specific to mapping services, then it should not be done there.

Adding an argument to the displaymap parser function can be done in `getParameterInfo` of `MapsDisplayMap`. If you want the parameter to also be available in ask queries, then you can stuff it in `getCommonMapParams` there. Though perhaps it's better to just add it in `SMMapPrinter`.

**Pull request merged**